### PR TITLE
add blog about ed25519 keypairs and openssl

### PR DESCRIPTION
**Pull request by sbtaylor15**

### CODE REVIEW

File: blog/2023-02-01-rust-ed25519-key-pairs.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  image_url: https://github.com/sbtaylor15.png
+tags: [rust, openssl, ed25519, keypair]
+---
+
+The ED25519 crate (libp2p::identity) can generate and load key pairs.  These key pairs are 64 bytes long.
+First 32 bytes are the private key and last 32 bytes the public key.
+
+Openssl generates the ED25519 key pairs with a header in front of the public and private keys.
+
+As result, the Openssl keys pairs and Rust key pairs are not interchangable.  Therefor, if you want to use
```
Comment (fishseabowl) [inline blog/2023-02-01-rust-ed25519-key-pairs.md:15]: typo
```suggestion
As a result, the Openssl keys pairs and Rust key pairs are not interchangeable. Therefor, if you want to use
```

File: blog/2023-02-01-rust-ed25519-key-pairs.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+---
+title: Openssl and Rust ED25519 Key Pairs
```
Comment (fishseabowl) [inline blog/2023-02-01-rust-ed25519-key-pairs.md:2]: I guess there are other ed25519 providers in Rust. Maybe libp2p rust or ed25519_dalek are more accurate

**Review by betarelease — APPROVED** [review]
This is good to document anyway. We should probably have an automation that allows people to just go back and forth between these formats. I will create a ticket.

File: blog/2023-02-01-rust-ed25519-key-pairs.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Below are the steps needed to convert key pairs from one format to another.
+
+## Generate a Key Pair
+
+```rust
+let keypair = identity::ed25519::Keypair::generate();
+```
+
+## Load a Key Pair
```
Comment (dasmanas) [inline blog/2023-02-01-rust-ed25519-key-pairs.md:27]:
```suggestion
## Load Openssl generated Key Pair in Rust Object
```

File: blog/2023-02-01-rust-ed25519-key-pairs.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+```rust
+let mut keypair_file = fs::File::open("ed25519.ser")?;
+let mut buffer = vec![0; 64];  // buffer containing the key pair from disk
+keypair_file.read_exact(&mut buffer)?; // read the file into the buffer
+let keypair = identity::ed25519::Keypair::decode(&mut buffer); // decode the buffer into a Keypair object
+```
```
Comment (dasmanas) [inline blog/2023-02-01-rust-ed25519-key-pairs.md:34]: Should this block to be moved at line 70 i.e. just after the ed25519.ser file generated through openssl and some post processing for a better sequence?

**Review by fishseabowl — APPROVED** [review]

**Pull request merged**